### ISSUE TITLE
Different highlighting for objective-C type argument references in method declarations

### ISSUE THREAD

**Issue by MihalisL** [tmont/sunlight#7]
Sorry to be a pain once again, but highlighting argument references in method declarations differs in handling from basic types to objective-C types.

Having followed site instructions for sunlight.js and highlighting the same article as yesterday, I get the following in a method declaration:

![screen shot 2014-01-06 at 21 08 45](https://f.cloud.github.com/assets/660029/1853060/0304f460-7706-11e3-9152-30789b6533c7.png)

"viewControllerClass" should be black like its "animated" counterpart. The same happens here too:

![screen shot 2014-01-06 at 21 10 36](https://f.cloud.github.com/assets/660029/1853073/4380ef94-7706-11e3-97c0-17394c577f72.png)

but "index" is coloured correctly within the method braces. 

I do not think this is the intended functionality as in the demo page of the sunlight.js site objective-C highlighting works differently:

![screen shot 2014-01-06 at 21 13 15](https://f.cloud.github.com/assets/660029/1853104/c2f2f5ba-7706-11e3-9b54-ae532103ee8c.png)

Thank you.

**Comment by tmont** [comment]
Hmmm... interesting. Not really sure what's going on there, and I can't reproduce it.

Is that the entire snippet of code you're trying to highlight? If not, can you paste it in? Or use https://gist.github.com/ if it's really long.

**Comment by tmont** [comment]
Never mind I reproduced it. I'll look into it.

**Comment by tmont** [comment]
This has been fixed in v1.22.0. You can download the pre-minified version from the [website](http://sunlightjs.com/#download) or get the source code from the [release tag](https://github.com/tmont/sunlight/releases/tag/1.22.0).

The reason you didn't see this bug manifest itself in the demo was because the demo used a pointer (`(ClassName *)ident`), which fooled the parser into thinking the ident should not be highlighted (a `named-ident`). Now it's slightly smarter, in that it looks ahead to make sure there isn't a colon following the following ident, which indicates it's a typed argument reference.

So before something like `(Baz)foo bar:Type` would highlight `foo` because it is preceded by another ident (ignoring the parentheses). Now it looks ahead to make sure it's not followed by a colon.

In short, the parser was too naive, and now it's slightly less naive :)

Thanks for the bug report.

**Comment by MihalisL** [comment]
This parser is not naive at all. :)

Great stuff! Thank you very much.
